### PR TITLE
fixes #14917; Consider the case where `nkBracket` is of type `tySequence`

### LINKED ISSUE #14917: Static seqs are turned into static arrays when passed through template.
### Example
```nim
proc passSeq2[T](a : static seq[T]) =
  echo a

template passSeq1[T](a : static seq[T]) =
  static:
    echo type(a)
  passSeq2(a)

passSeq1(@[5,6])
```

### Current Output

```
array[0..1, int]
testNim.nim(11, 9) Error: type mismatch:
got <static[array[0..1, int]]([5, 6])>
but expected one of:
proc passSeq2[T](a: static seq[T])
  first type mismatch at position: 1
  required type for a: static[seq[T]]
  but expression '[5, 6]' is of type: static[array[0..1, int]]([5, 6])

expression: passSeq2([5, 6])
```
### Expected Output
```
seq[int]
@[5,6]
```
```
$ nim -v
Nim Compiler Version 1.2.2 [Windows: amd64]
```

### PR DESCRIPTION
**Pull request by Pylgos**
fixes #14917

### CODE REVIEW

File: compiler/semexprs.nim [hunk 1]
```diff
@@ -643,6 +643,15 @@ proc arrayConstrType(c: PContext, n: PNode): PType =
   result = typ
 
 proc semArrayConstr(c: PContext, n: PNode, flags: TExprFlags; expectedType: PType = nil): PNode =
+  if n.typ != nil and n.typ.kind == tySequence:
```
Comment (metagn) [inline compiler/semexprs.nim]: Sorry if this is outside your comfort range, but could we maybe wait until the end and sem the array as normal and finally change the type to `seq` from the formed `array` type if the node was originally typed as `seq`? Just curious if it accomplishes the same thing.

**Comment by Araq** [comment]
This doesn't fix the root cause.

**Comment by ringabout** [comment]
I stumbled upon the same problem, just for the record: https://github.com/nim-lang/Nim/pull/20662

**Comment by metagn** [comment]
What is the root cause? Is the VM generating `nkBracket` bad?

**Comment by Araq** [comment]
I don't know, but I do know that blindly reseming expressions is not a good solution.

**Pull request closed without merging**